src/rotation.rs

```diff
@@ -0,0 +1,645 @@
+// Copyright 2013 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use approxeq::ApproxEq;
+use num_traits::{Float, One, Zero};
+use std::fmt;
+use std::ops::{Add, Neg, Mul, Sub, Div};
+use std::marker::PhantomData;
+use trig::Trig;
+use {TypedPoint2D, TypedPoint3D, TypedVector2D, TypedVector3D, Vector3D, point2, point3, vec3};
+use {TypedTransform3D, TypedTransform2D, UnknownUnit, Radians};
+
+
+define_matrix! {
+    /// A transform that can represent rotations in 2d, represented as an angle in radians.
+    pub struct TypedRotation2D<T, Src, Dst> {
+        pub angle : T,
+    }
+}
+
+/// The default 2d rotation type with no units.
+pub type Rotation2D<T> = TypedRotation2D<T, UnknownUnit, UnknownUnit>;
+
+impl<T, Src, Dst> TypedRotation2D<T, Src, Dst> {
+    #[inline]
+    /// Creates a rotation from an angle in radians.
+    pub fn new(angle: Radians<T>) -> Self {
+        TypedRotation2D {
+            angle: angle.0,
+            _unit: PhantomData,
+        }
+    }
+
+    /// Creates the identity rotation.
+    #[inline]
+    pub fn identity() -> Self where T: Zero {
+        Self::new(Radians::new(T::zero()))
+    }
+}
+
+impl<T, Src, Dst> TypedRotation2D<T, Src, Dst> where T: Clone
+{
+    /// Returns self.angle as a strongly typed `Radians<T>`.
+    pub fn get_angle(&self) -> Radians<T> {
+        Radians::new(self.angle.clone())
+    }
+}
+
+
+impl<T, Src, Dst> TypedRotation2D<T, Src, Dst>
+where T: Copy + Clone +
+         Add<T, Output=T> +
+         Sub<T, Output=T> +
+         Mul<T, Output=T> +
+         Div<T, Output=T> +
+         Neg<Output=T> +
+         ApproxEq<T> +
+         PartialOrd +
+         Float +
+         One + Zero
+{
+    /// Creates a 3d rotation (around the z axis) from this 2d rotation.
+    #[inline]
+    pub fn to_3d(&self) -> TypedRotation3D<T, Src, Dst> {
+        TypedRotation3D::around_z(self.get_angle())
+    }
+
+    /// Returns the inverse of this rotation.
+    #[inline]
+    pub fn inverse(&self) -> TypedRotation2D<T, Dst, Src> {
+        TypedRotation2D::new(-self.get_angle())
+    }
+
+    /// Returns the given 2d point transformed by this rotation.
+    ///
+    /// The input point must be use the unit Src, and the returned point has the unit Dst.
+    #[inline]
+    pub fn transform_point(&self, point: &TypedPoint2D<T, Src>) -> TypedPoint2D<T, Dst> {
```

Comment:
why is it `transform_point` as opposed to `rotate_point`? I'd expect `transform_*` to be reserved for typed transformations

Reply:
I hadn't thought about it. `rotate_point` indeed makes sense.